src/window-event-handler.coffee

```diff
@@ -24,13 +23,12 @@ class WindowEventHandler
             if pathToOpen? and needsProjectPaths
               if fs.existsSync(pathToOpen)
                 atom.project.addPath(pathToOpen)
-              else if fs.existsSync(path.dirname(pathToOpen))
-                atom.project.addPath(path.dirname(pathToOpen))
               else
-                atom.project.addPath(pathToOpen)
+                dirToOpen = path.dirname(pathToOpen)
+                if fs.existsSync(dirToOpen)
+                  atom.project.addPath(dirToOpen)
 
-            {protocol} = url.parse(pathToOpen)
-            unless fs.isDirectorySync(pathToOpen) or protocol?
+            unless fs.isDirectorySync(pathToOpen)
```

Comment:
@mostafaeweda The 2 problems we found that this PR introduced were both in this file:

1. `pathToOpen` can be `undefined`, if we're trying to open a new empty editor.
2. If `pathToOpen` is a URI, we still need to open it.